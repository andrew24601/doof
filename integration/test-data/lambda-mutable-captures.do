// Integration test validating mutable lambda captures via wrapper objects

function updateCapturedLocal(): void {
    let localCounter: int = 5;

    const apply = ():int => {
        localCounter += 3;
        return localCounter;
    };

    println(localCounter);
    println(apply());
    println(localCounter);
}

function makeIncrementer(start: int): (): int {
    let current: int = start;
    return => {
        current++;
        return current;
    };
}

function main(): int {
    updateCapturedLocal();

    let increment = makeIncrementer(10);
    println(increment());
    println(increment());

    return 0;
}
