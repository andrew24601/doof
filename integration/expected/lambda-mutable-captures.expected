5
8
8
11
12

